js/packages/openinference-instrumentation-llama-index/src/instrumentation.ts

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type * as llamaindex from "llamaindex";
+
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+  InstrumentationModuleDefinition,
+  InstrumentationNodeModuleDefinition,
+  safeExecuteInTheMiddle,
+} from "@opentelemetry/instrumentation";
+import { VERSION } from "./version";
+import {
+  Span,
+  SpanKind,
+  SpanStatusCode,
+  context,
+  diag,
+  trace,
+} from "@opentelemetry/api";
+import { isTracingSuppressed } from "@opentelemetry/core";
+
+const MODULE_NAME = "llamaindex";
+
+/**
+ * Flag to check if the openai module has been patched
+ * Note: This is a fallback in case the module is made immutable (e.x. Deno, webpack, etc.)
+ */
+let _isOpenInferencePatched = false;
+
+/**
+ * function to check if instrumentation is enabled / disabled
+ */
+export function isPatched() {
+  return _isOpenInferencePatched;
+}
+
+import {
+  MimeType,
+  OpenInferenceSpanKind,
+  SemanticConventions,
+} from "@arizeai/openinference-semantic-conventions";
+
+/**
+ * Resolves the execution context for the current span
+ * If tracing is suppressed, the span is dropped and the current context is returned
+ * @param span
+ */
+function getExecContext(span: Span) {
+  const activeContext = context.active();
+  const suppressTracing = isTracingSuppressed(activeContext);
+  const execContext = suppressTracing
+    ? trace.setSpan(context.active(), span)
+    : activeContext;
+  // Drop the span from the context
+  if (suppressTracing) {
+    trace.deleteSpan(activeContext);
+  }
+  return execContext;
+}
+export class LlamaIndexInstrumentation extends InstrumentationBase<
+  typeof llamaindex
+> {
+  constructor(config?: InstrumentationConfig) {
+    super(
+      "@arizeai/openinference-instrumentation-llama-index",
+      VERSION,
+      Object.assign({}, config),
+    );
+  }
+  manuallyInstrument(module: typeof llamaindex) {
+    diag.debug(`Manually instrumenting ${MODULE_NAME}`);
+    this.patch(module);
+  }
```

Comment:
doesn't look like we need this, are you just adding it for flexibility?

Reply:
In general I need this from now on to have an escape hatch for ESM / NextJS